autoblocks/_impl/testing/evaluators/ragas_faithfulness.py

```diff
@@ -0,0 +1,21 @@
+import abc
+from typing import Generic
+
+from autoblocks._impl.testing.evaluators.ragas_base import BaseRagas
+from autoblocks._impl.testing.models import Evaluation
+from autoblocks._impl.testing.models import OutputType
+from autoblocks._impl.testing.models import TestCaseType
+
+
+class BaseRagasFaithfulness(BaseRagas[TestCaseType, OutputType], abc.ABC, Generic[TestCaseType, OutputType]):
+    """
+    The RagasFaithfulness evaluator evaluates the factual consistency of the generated answer against the given context.
+
+    See more: https://docs.ragas.io/en/stable/concepts/metrics/faithfulness.html
+    """
+
+    def evaluate_test_case(self, test_case: TestCaseType, output: OutputType) -> Evaluation:
```

Comment:
it looks like `evaluate_test_case` can live on `BaseRagas`, you just need to specify the two things that are changing on each subclass:

```python
class BaseRagasFaithfulness(
  BaseRagas[TestCaseType, OutputType],
  abc.ABC,
  Generic[TestCaseType, OutputType],
):
  ragas_metrics = [ragas.metrics.faithfulness]
  result_name = "faithfulness"

class BaseRagasContextRelevancy(
  BaseRagas[TestCaseType, OutputType],
  abc.ABC,
  Generic[TestCaseType, OutputType],
):
  ragas_metrics = [ragas.metrics.context_relevancy]
  result_name = "context_relevancy"
```

etc

then in `BaseRagas`:

```python
    def evaluate_test_case(self, test_case: TestCaseType, output: OutputType) -> Evaluation:
        dataset = self.make_dataset(test_case=test_case, output=output)
        ragas = self.get_ragas()  # type: ignore[no-untyped-call]
        result = ragas.evaluate(dataset=dataset, metrics=self.ragas_metrics)
        return Evaluation(score=result[self.result_name], threshold=self.threshold)
```

Reply:
Is that going to work without importing ragas in the evaluator?

Reply:
oh i see, yeah you could access by name if it's always the same:

```python
class BaseRagasFaithfulness(
  BaseRagas[TestCaseType, OutputType],
  abc.ABC,
  Generic[TestCaseType, OutputType],
):
  metric_name = "faithfulness"
```

```python
        dataset = self.make_dataset(test_case=test_case, output=output)
        ragas = self.get_ragas()  # type: ignore[no-untyped-call]
        result = ragas.evaluate(dataset=dataset, metrics=[getattr(ragas.metrics, self.metric_name)])
        return Evaluation(score=result[self.metric_name], threshold=self.threshold)
```